recipes/tk/all/conanfile.py

```diff
@@ -48,66 +66,84 @@ def _settings_build(self):
+        if not is_msvc(self):
+            if (
+                self._settings_build.os == "Windows"
+                and not self.conf.get("tools.microsoft.bash:path")
+                and not self.conf.get("tools.microsoft.bash:subsystem")
+            ):
                 self.build_requires("msys2/cci.latest")
 
     def validate(self):
-        if self.options["tcl"].shared != self.options.shared:
-            raise ConanInvalidConfiguration("The shared option of tcl and tk must have the same value")
-
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  strip_root=True, destination=self._source_subfolder)
-
-    def _patch_sources(self):
-        for build_system in ("unix", "win", ):
-            config_dir = self._get_configure_folder(build_system)
-
-            if build_system != "win":
-                # When disabling 64-bit support (in 32-bit), this test must be 0 in order to use "long long" for 64-bit ints
-                # (${tcl_type_64bit} can be either "__int64" or "long long")
-                tools.replace_in_file(os.path.join(config_dir, "configure"),
-                                      "(sizeof(${tcl_type_64bit})==sizeof(long))",
-                                      "(sizeof(${tcl_type_64bit})!=sizeof(long))")
-
-            makefile_in = os.path.join(config_dir, "Makefile.in")
-            # Avoid clearing CFLAGS and LDFLAGS in the makefile
-            # tools.replace_in_file(makefile_in, "\nCFLAGS{}".format(" " if (build_system == "win" and name == "tcl") else "\t"), "\n#CFLAGS\t")
-            tools.replace_in_file(makefile_in, "\nLDFLAGS\t", "\n#LDFLAGS\t")
-            tools.replace_in_file(makefile_in, "${CFLAGS}", "${CFLAGS} ${CPPFLAGS}")
-
-        rules_ext_vc = os.path.join(self.source_folder, self._source_subfolder, "win", "rules-ext.vc")
-        tools.replace_in_file(rules_ext_vc,
-                              "\n_RULESDIR = ",
-                              "\n_RULESDIR = .\n#_RULESDIR = ")
-        rules_vc = os.path.join(self.source_folder, self._source_subfolder, "win", "rules.vc")
-        tools.replace_in_file(rules_vc,
-                              r"$(_TCLDIR)\generic",
-                              r"$(_TCLDIR)\include")
-        tools.replace_in_file(rules_vc,
-                              "\nTCLSTUBLIB",
-                              "\n#TCLSTUBLIB")
-        tools.replace_in_file(rules_vc,
-                              "\nTCLIMPLIB",
-                              "\n#TCLIMPLIB")
+        if self.dependencies["tcl"].options.shared != self.options.shared:
+            raise ConanInvalidConfiguration(
+                "The shared option of tcl and tk must have the same value"
+            )
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            raise ConanInvalidConfiguration("The tk conan recipe does not currently support Macos M1 builds. A contribution to add this functionality would be welcome.")
```

Comment:
It seems macOS m1 builds are cross-builds on CCI. A local build on M1 works just fine. I tried to cross-compile x86_64 from M1 and it fails.
```suggestion
        if self.settings.os == "Macos" and cross_building(self):
            raise ConanInvalidConfiguration("The tk conan recipe does not currently support Macos cross-builds. A contribution to add this functionality would be welcome.")
```

Reply:
cross-build fixed by https://github.com/conan-io/conan-center-index/pull/19309/commits/e7d3358a0a8dde9d80824050ecd59d0fee721e43

I'm not sure that's the correct way to do this but it works (at least locally, currently building on CCI CI).

Reply:
@mayeut thanks for that. Happy to keep that out of scope in this PR and fix it on yours if the Conan team are too